frontend/test/pytest/test_peephole_optimizations.py

```diff
@@ -151,10 +151,10 @@ def test_pipeline_functionality(theta, backend):
     Test that the @pipeline decorator does not change functionality
     when all the passes in the pipeline does not change functionality.
     """
-    my_pipeline = {
-        "cancel_inverses": {},
-        "merge_rotations": {},
-    }
+    my_pipeline = [
+        Pass("remove-chained-self-inverse"),
+        Pass("merge-rotations"),
+    ]
```

Comment:
@erick-xanadu what is the motivation for making this change? I must admit as a user I find the one on the left much more usable (configuration that requires me to import additional objects, and read more documentation/docstrings complicates the process).

Could we not keep the dictionary user-facing, but apply `Pass` internally?

Reply:
It is a bit subjective in my opinion. This change I believe is a better design because:

1. In the previous case where there are no options, we still needed to create an empty dictionary. In this case, options can be passed as arguments to the Pass object.
2. In the previous case there was no nice syntax to specify options without a required value. I.e., if I have an option like `--option` instead of `--option-value=value`, I need to create an entry in the option dictionary and have a `None` in it? In this case, all options are pythonic. `*args` are options without values and `**kwargs` are options with values.
3. We also remove the need of having this mapping between strings and API.

> and read more documentation/docstrings complicates the process

The user still needs to read the documentation to know which options exist. Otherwise what sort of dictionary will they create? This at least formalizes it a little bit more with the ability to check at runtime whether the options are valid for each pass. (I.e., I can create a class that inherits from Pass and checks that the required options are set by the user).

Reply:
Decision: keep dictionary approach for now, but add in

```
my_pipeline = [
    cancel_inverses(*options, **kwargs),
    Pass("another-mlir-named-pass"),
    merge_rotations(**kwargs),
]
```

Future considerations:

- unifying our UI with PLs.

Reply:
I've converted the dictionary to a tuple of passes. I haven't updated the interface to allow for the list / tuple above, but that can come in a later PR. For now, the infrastructure is ready to implement it.